engine/src/main/java/org/terasology/engine/subsystem/rpc/DiscordRPCSubSystem.java

```diff
@@ -0,0 +1,308 @@
+    private static final long CLIENT_ID = 515274721080639504L;
+    private static final String LARGE_IMAGE = "ss_6";
+    private static DiscordRPCSubSystem instance;
+
+    private IPCClient ipcClient;
+    private boolean ready;
+    private boolean autoReconnect;
+    private Thread reconnectThread;
+    private RichPresence lastRichPresence;
+    private boolean reconnecting;
+    private int reconnectTries = 1;
+    private boolean connectedBefore;
+    private int lastPing;
+    private Config config;
+    private String lastState;
+    private boolean dontTryAgain;
+    private boolean enabled;
+
+    public DiscordRPCSubSystem() throws IllegalStateException {
+        if (instance != null) {
+            throw new IllegalStateException("More then one instance in the DiscordRPC");
+        }
+        lastRichPresence = null;
+        ipcClient = new IPCClient(CLIENT_ID);
+        ipcClient.setListener(this);
+        autoReconnect = true;
+        reconnectThread = new Thread(this);
+        reconnectThread.setName("DISCORD-RPC-RECONNECT");
+        reconnectThread.start();
+        instance = this;
+        enabled = false;
+        dontTryAgain = true;
+    }
+
+    public void sendRichPresence(RichPresence richPresence) {
+        this.lastRichPresence = richPresence;
+        if (!ready || lastRichPresence == null || !enabled) {
+            return;
+        }
+        ipcClient.sendRichPresence(lastRichPresence);
+    }
+
+    @Override
+    public void onReady(IPCClient client) {
+        if (reconnecting) {
+            getLogger().info("Discord RPC >> Reconnected!");
+            reconnectTries = 1;
+        } else {
+            getLogger().info("Discord RPC >> Connected!");
+            connectedBefore = true;
+        }
+        this.ipcClient = client;
+        if (!ready) {
+            ready = true;
+        }
+        if (lastRichPresence == null) {
+            RichPresence.Builder builder = new RichPresence.Builder();
+            builder.setLargeImage(LARGE_IMAGE);
+            lastRichPresence = builder.build();
+        }
+        client.sendRichPresence(lastRichPresence);
+    }
+
+    @Override
+    public void onDisconnect(IPCClient client, Throwable t) {
+        if (ready) {
+            ready = false;
+        }
+        getLogger().info("Discord RPC >> Disconnected!");
+    }
+
+    @Override
+    public void run() {
+        while (autoReconnect) {
+            try {
+                // Ignore if the Discord RPC is not enabled
+                if (!enabled) {
+                    if (ready) {
+                        try {
+                            getInstance().ipcClient.close();
+                        } catch (Exception ex) { }
+                    }
+                    Thread.sleep(1);
+                    continue;
+                }
+
+                // Don't try to retry to discover the RPC until some reasons happen
```

Comment:
This comment could probably be a bit more specific :-)

Reply:
**Done** I give it more special meaning than before. 😄